tools/ci_build/github/azure-pipelines/post-merge-jobs.yml

```diff
@@ -1,30 +1,39 @@
+variables:
+- name: WIN_CPU_BUILD_MACHINE_POOL_NAME  # name of a variable
+  # The public ADO project
+  ${{ if eq(variables['System.CollectionId'], 'f3ad12f2-e480-4533-baf2-635c95467d29') }}:
+    value: 'Win-CPU-2019'
+  # The private ADO project
+  ${{ if eq(variables['System.CollectionId'], 'bc038106-a83b-4dab-9dd3-5a41bc58f34c') }}:
+    value: 'Win-CPU-2021'
+- name: LINUX_CPU_BUILD_MACHINE_POOL_NAME  # name of a variable
+  # The public ADO project
+  ${{ if eq(variables['System.CollectionId'], 'f3ad12f2-e480-4533-baf2-635c95467d29') }}:
+    value: 'Linux-CPU-2019'
+  # The private ADO project
+  ${{ if eq(variables['System.CollectionId'], 'bc038106-a83b-4dab-9dd3-5a41bc58f34c') }}:
+    value: 'Linux-CPU'
```

Comment:
Awesome!

Could we go one step further so the obtuse collection id guid is only used in one place? Something like

```yaml
variables:
- name: BUILD_LOCATION  # name of a variable
  # The public ADO project
  ${{ if eq(variables['System.CollectionId'], 'f3ad12f2-e480-4533-baf2-635c95467d29') }}:
    value: 'GitHub'
  # The private ADO project
  ${{ if eq(variables['System.CollectionId'], 'bc038106-a83b-4dab-9dd3-5a41bc58f34c') }}:
    value: 'AIInfra'

variables:
- name: WIN_CPU_BUILD_MACHINE_POOL_NAME  # name of a variable
  # The public ADO project
  ${{ if eq(variables['BUILD_LOCATION'], 'GitHub') }}:
    value: 'Win-CPU-2019'
  # The private ADO project
  ${{ if eq(variables['BUILD_LOCATION'], 'AIInfra') }}:
    value: 'Win-CPU-2021'
- name: LINUX_CPU_BUILD_MACHINE_POOL_NAME  # name of a variable
  # The public ADO project
  ${{ if eq(variables['BUILD_LOCATION'], 'GitHub') }}:
    value: 'Linux-CPU-2019'
  # The private ADO project
  ${{ if eq(variables['BUILD_LOCATION'], 'AIInfra') }}:
    value: 'Linux-CPU'    
```

Or alternatively just name the collection id

```yaml
variables:
# The public ADO project
- name: GitHubCollectionId  # name of a variable
  value: 'f3ad12f2-e480-4533-baf2-635c95467d29'
  # The private ADO project
- name: AIInfraCollectionId
  value: 'bc038106-a83b-4dab-9dd3-5a41bc58f34c'

variables:
- name: WIN_CPU_BUILD_MACHINE_POOL_NAME  # name of a variable
  # The public ADO project
  ${{ if eq(variables['System.CollectionId'], variables['GitHubCollectionId']) }}:
    value: 'Win-CPU-2019'
  # The private ADO project
  ${{ if eq(variables['System.CollectionId'], variables['AIInfraCollectionId']) }}:
    value: 'Win-CPU-2021'
- name: LINUX_CPU_BUILD_MACHINE_POOL_NAME  # name of a variable
  # The public ADO project
  ${{ if eq(variables['System.CollectionId'], variables['GitHubCollectionId']) }}:
    value: 'Linux-CPU-2019'
  # The private ADO project
  ${{ if eq(variables['System.CollectionId'], variables['AIInfraCollectionId']) }}:
    value: 'Linux-CPU'
```